azurerm/internal/services/cosmos/cosmosdb_sql_container_resource.go

```diff
@@ -158,9 +166,17 @@ func resourceArmCosmosDbSQLContainerCreate(d *schema.ResourceData, meta interfac
 	}
 
 	if partitionkeypaths != "" {
-		db.SQLContainerCreateUpdateProperties.Resource.PartitionKey = &documentdb.ContainerPartitionKey{
-			Paths: &[]string{partitionkeypaths},
-			Kind:  documentdb.PartitionKindHash,
+		if hasPartitionKeyVersion {
+			db.SQLContainerCreateUpdateProperties.Resource.PartitionKey = &documentdb.ContainerPartitionKey{
+				Paths:   &[]string{partitionkeypaths},
+				Kind:    documentdb.PartitionKindHash,
+				Version: utils.Int32(int32(partitionKeyVersion.(int))),
+			}
+		} else {
+			db.SQLContainerCreateUpdateProperties.Resource.PartitionKey = &documentdb.ContainerPartitionKey{
+				Paths: &[]string{partitionkeypaths},
+				Kind:  documentdb.PartitionKindHash,
+			}
 		}
 	}
```

Comment:
We could make this something like
```suggestion
	if partitionkeypaths != "" {
		db.SQLContainerCreateUpdateProperties.Resource.PartitionKey = &documentdb.ContainerPartitionKey{
			Paths: &[]string{partitionkeypaths},
			Kind:  documentdb.PartitionKindHash,
		}
		
		if v, ok := d.GetOk("partition_key_version"); ok {
			db.SQLContainerCreateUpdateProperties.Resource.PartitionKey.Version = utils.Int32(int32(partitionKeyVersion.(int))),
		}
	}
```

Reply:
resolved